### ISSUE TITLE
Clarify the distinction between `sct_image` and `sct_maths`

### ISSUE THREAD

**Issue by joshuacwnewton** [spinalcordtoolbox/spinalcordtoolbox#4039]
### Background

SCT has two similar CLI scripts: `sct_image` and `sct_maths`. Both scripts follow roughly the same structure: They are a collection of distinct, individual operations, where each CLI flag will perform a different operation on the image.

### Issue

Due to this similarity, I sometimes get a bit confused. ("Should this option be in `sct_image` or `sct_maths`? Where should I look?")

The argparse help descriptions don't exactly help distinguish between the two scripts:

```bash
# sct_image
Perform manipulations on images (e.g., pad, change space, split along dimension). Inputs can be a
number, a 4d image, or several 3d images separated with ","

# sct_maths
Perform mathematical operations on images.
```

Since both scripts modify the data array of an image, where do we draw the line?

### Proposed solution

Some observations:

* `sct_image` has options such as `-pad`, `-split`, `-remove-vol`, and `-setorient`.

   These operations manipulate the _structure_ of the image. They may modify the header and/or the array, but they treat the actual _contents_ of the data array as a black box. In other words, there are no changes to individual voxels.
 
* `sct_maths` has options such as `-add`, `-mean`, `-dilate`, `-otsu`, and `-denoise`.

   These operations modify the _values_ of individual voxels inside the data array.

Assuming this distinction is correct, precise, and applies to all the existing options, then we should consider updating the argparse help descriptions to clarify the distinction:

```yaml
# sct_image
Modifies the header and/or array of an image. 

NB: The operations in this script are general-purpose, and so they do not modify individual 
voxels in the data array. To perform mathematical operations on specific voxels in the data 
array, please refer to the `sct_maths` script.

# sct_maths
Performs mathematical operations on individual voxels in an image's data array.

NB: The operations in this script are data-specific. To perform general-purpose operations 
on an image's header and array, please refer to the `sct_image` script.
```

This is just a rough draft, might be a bit wordy, so there's probably a way to edit this to be simpler. But, hopefully the gist is clear.

**Comment by valosekj** [comment]
Great initiative!

## sct_maths

I understand `sct_maths` as an analogy to [`fslmaths`](https://fsl.fmrib.ox.ac.uk/fsl/fslwiki/Fslutils):

> simple but powerful program to allow mathematical manipulation of images.

This corresponds with the `sct_maths` description:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/158abd122c134c4cb9b447bc8623c75a3c9bf8e2/spinalcordtoolbox/scripts/sct_maths.py#L68

I agree with you that:

> These operations modify the values of individual voxels inside the data array.

In other words, the dimensions of the input image remain the same.

## sct_image

Again, I agree with you:

> These operations manipulate the structure of the image. They may modify the header and/or the array, but they treat the actual contents of the data array as a black box. In other words, there are no changes to individual voxels.

In other words, the dimensions/header of the input image is changed. Which is implicitly stated in `sct_image` description:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/158abd122c134c4cb9b447bc8623c75a3c9bf8e2/spinalcordtoolbox/scripts/sct_image.py#L34-L35

Maybe, as you proposed, we could add explicit information that the header and/or array is modified:

> ```
> Modifies the header and/or array of images. 
> ```

**Comment by jcohenadad** [comment]
> In other words, the dimensions of the input image remain the same.

not always, eg `sct_maths -i INPUT4D -mean t -o OUTPUT3D`
